rollup.config.js

```diff
@@ -0,0 +1,93 @@
+import babel from '@rollup/plugin-babel';
+import typescript from '@rollup/plugin-typescript';
+import { terser } from 'rollup-plugin-terser';
+
+/** @type {import("rollup").RollupOptions} */
+const defaultConfig = {
+  input: 'src/index.ts',
+  external: [/@babel\/runtime/, /@reduxjs\/toolkit/, /react/, /react-redux/, /immer/, /tslib/],
+  treeshake: {
+    propertyReadSideEffects: false,
+  },
+};
+
+const defaultTerserOptions = {
+  output: { comments: false },
+  compress: {
+    keep_infinity: true,
+    pure_getters: true,
+    passes: 10,
+  },
+  ecma: 5,
+  warnings: true,
+};
+
+const defaultTsConfig = { declaration: false, declarationMap: false, target: 'ESNext', module: 'esnext' };
+
+/** @type {import("rollup").RollupOptions} */
+const configs = [
+  // ESM
+  {
+    ...defaultConfig,
+    output: [
+      {
+        dir: 'dist/esm',
+        format: 'es',
+        sourcemap: true,
+        preserveModules: true,
+      },
+    ],
+    plugins: [
+      typescript(defaultTsConfig),
+      babel({
+        exclude: 'node_modules/**',
+        extensions: ['.js', '.ts'],
+        babelHelpers: 'runtime',
+        presets: [
+          [
+            '@babel/preset-env',
+            {
+              targets: { node: true, browsers: ['defaults', 'not IE 11', 'maintained node versions'] },
```

Comment:
So this now has two builds: an ESM build that does not bundle up the transpiled files, so just a file-for-file transpilation. This should help modern bundlers with tree-shaking quite a lot.
The target I chose for that is "all non-dead browsers, last 2 versions, no IE, maintained node versions".

The CJS build includes IE as well and builds only one file.

Do you think that's sensible @markerikson ?

Reply:
For reference: including IE there bumps the bundle sizes for the ESM module up to

```
  ESM full
  Size limit: 15 KB
  Size:       13.8 KB with all dependencies, minified and gzipped
  
  createApi + setupListeners
  Size: 12.55 KB with all dependencies, minified and gzipped
  
  fetchBaseQuery
  Size: 3.23 KB with all dependencies, minified and gzipped
  
  retry
  Size: 2.79 KB with all dependencies, minified and gzipped
  
  ApiProvider
  Size: 387 B with all dependencies, minified and gzipped
```

Reply:
Sure, fine by me.